apps/frontend/client/src/lib/data/lpc_asset_catalog.ts

```
@@ -76,6 +76,43 @@ export const DIRECTION_OPTIONS: readonly { value: LpcDirection; label: string }[
+ * Returns an empty catalog if the seed is not yet loaded.
+ */
+export const getLpcCatalog = (): LpcCatalog => {
+  const seed = assetStore.seed;
+  if (!seed) {
+    return { slots: [], assetIdsBySlot: {}, allAssetIds: [] };
+  }
+  if (_lastSeed !== seed.rows) {
+    _lastSeed = seed.rows;
+    _cachedCatalog = buildLpcCatalog({ entries: seed.rows });
+  }
+  return _cachedCatalog!;
```

Biome (2.5.7)

[error] Forbidden non-null assertion.

(lint/style/noNonNullAssertion)